R/textstat_lexdiv.R

```diff
@@ -286,18 +286,24 @@ dfm_split_hyphenated_features <- function(x) {
     
     # figure out where the hyphens are
     hyphenated_index <- which(stringi::stri_detect_regex(featnames(x), hyphen_regex))
-
-    # split the hyphenated feature names into a list of components
-    splitfeatures <- as.list(tokens(featnames(x)[hyphenated_index], remove_hyphens = TRUE))
-    
-    # efficiently create a new dfm from hyphenated feature name components
-    splitdfm <- x[, rep(hyphenated_index, times = lengths(splitfeatures))]
-    colnames(splitdfm) <- unlist(splitfeatures, use.names = FALSE)
     
-    # combine dfms and suppress duplicated feature name warning
-    result <- suppressWarnings(cbind(x[, -hyphenated_index], splitdfm))
-    # compress features to combine same-named features
-    result <- dfm_compress(result, margin = "features")
-    
-    result
+    # Add check if there are any hyphenated features at all
+    if (!(length(hyphenated_index) == 0)){
```

Comment:
Good catch. I changed this to just return(x) if no hyphenated features are found.
Also the return is usually just an evaluated expression. See http://adv-r.had.co.nz/Functions.html#return-values.